test/controllers/trips_controller_test.rb

```diff
@@ -2,22 +2,149 @@
 
 describe TripsController do
   describe "show" do
+    it "will get show for valid id" do
+      driver = Driver.create(name: "Marta", vin: "RF5J464C70D9C3KBT")
+      passenger = Passenger.create(name: "Leah", phone_num: "2060772909")
+      trip = Trip.create(driver_id: driver.id, passenger_id: passenger.id, date: Date.today, rating: 4, cost: 234)
+
+      valid_trip_id = trip.id
+      get "/trips/#{valid_trip_id}"
+      must_respond_with :success
+    end
+
+    it "will respond with not_found for invalid_id" do
+      invalid_trip_id = -1
+      get "/trips/#{invalid_trip_id}"
+      must_respond_with :not_found
+    end
     # Your tests go here
   end
 
   describe "create" do
-    # Your tests go here
+    it "can create a trip" do
+      driver = Driver.create(name: "Yeni", vin: "RF5J464C70D9C3KBT", available: "true")
+      passenger = Passenger.create(name: "Sam", phone_num: "2060772908")
+
+      trip_hash = {
+        trip: {
+          driver_id: driver.id,
+          passenger_id: passenger.id,
+          date: Date.today,
+          cost: 34,
+        },
+      }
+
+      expect {
+        post passenger_trips_path(passenger.id), params: trip_hash
+      }.must_differ "Trip.count", 1
+
+      expect(Trip.last.passenger_id).must_equal trip_hash[:trip][:passenger_id]
+
+      must_respond_with :redirect
+    end
+
+    # it "will not create a trip with invalid params" do
+    #   trip_hash = {
+    #     trip: {
+    #       driver_id: nil,
+    #       passenger_id: passenger.id,
+    #       date: Date.today,
+    #       cost: 45
+    #     }
+    #   }
+    #   expect {
+    #     post trips_path, params: trip_hash
+    #   }.wont_change "Trip.count"
+
+    #   must_respond_with :not_found
+    # end
+
   end
 
   describe "edit" do
+    it "succesfully edit and load" do
+      driver = Driver.create(name: "Liya", vin: "RF5J464C70D9C3KBT")
+      passenger = Passenger.create(name: "Calder", phone_num: "3456545678")
+      trip = Trip.create(driver_id: driver.id, passenger_id: passenger.id, date: Date.today, rating: 1, cost: 300)
+
+      get "/trips/#{trip.id}/edit"
```

Comment:
You should use named paths

```suggestion
      get edit_trip_path(trip.id)
```